hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/handle/record/RecordListBuilder.java

```diff
@@ -278,6 +289,14 @@ public void revertChildrenOf(@NonNull final SingleTransactionRecordBuilderImpl r
         for (int i = count - 1; i >= into; i--) {
             childRecordBuilders.remove(i);
         }
+
+        // If there are preceding child records somehow are added that are more than allowed preceding records,
+        // then we need to revert them as well.
+        // TODO: need to discuss if this is needed with team.
+        precedingTxnRecordBuilders.stream()
```

Comment:
I feel like we don't need the `skip(3)` here; and actually want destructive filtering that deletes the `removable()` record builders like,
```
precedingTxnRecordBuilders  = precedingTxnRecordBuilders
    .filter(Predicate.not(SingleTransactionRecordBuilderImpl::removable))
    .map(child -> child.status(child.status() == SUCCESS ? REVERTED_SUCCESS : child.status()))
    .toList();
```

Reply:
Makes sense ...Fixed